pytato/analysis/__init__.py

```diff
@@ -318,40 +323,39 @@ class DirectPredecessorsGetter(Mapper):
     of a node.
 
     .. note::
-
         We only consider the predecessors of a nodes in a data-flow sense.
     """
-    def _get_preds_from_shape(self, shape: ShapeType) -> frozenset[Array]:
-        return frozenset({dim for dim in shape if isinstance(dim, Array)})
+    def _get_preds_from_shape(self, shape: ShapeType) -> abc_Set[Array]:
```

Comment:
```suggestion
    def _get_preds_from_shape(self, shape: ShapeType) -> AbstractSet[Array]:
```
(from `typing`)? I'm not sure mypy understands the `collections.abc` types well.

Reply:
I think this may have been fixed with [076a76e](https://github.com/inducer/pytato/pull/529/commits/076a76ebe152f8d82c47de31a05f25586b981e4f)